spec/searchable_by_spec.rb

```diff
@@ -3,32 +3,37 @@
 describe ActiveRecord::SearchableBy do
   context 'norm_values' do
     def norm(str)
-      described_class.norm_values(str)
+      described_class.norm_values(str).each_with_object({}) do |val, acc|
+        acc[val.term] = val.negate
+      end
     end
 
     it 'should tokenise strings' do
-      expect(norm(nil)).to eq([])
-      expect(norm('""')).to eq([])
-      expect(norm('-+""')).to eq(%w[+ -])
-      expect(norm('simple words')).to eq(%w[simple words])
-      expect(norm(" with   \t spaces\n")).to eq(%w[with spaces])
-      expect(norm('with with duplicates with')).to eq(%w[with duplicates])
-      expect(norm('with "full term"')).to eq(['full term', 'with'])
-      expect(norm('"""odd double quotes around"""')).to eq(['odd double quotes around'])
-      expect(norm('""even double quotes around""')).to eq(['even double quotes around'])
-      expect(norm('with -"minus before"')).to eq(['-minus before', 'with'])
-      expect(norm('with "-minus within"')).to eq(['-minus within', 'with'])
-      expect(norm('with +"plus before"')).to eq(['+plus before', 'with'])
-      expect(norm('with "+plus within"')).to eq(['+plus within', 'with'])
-      expect(norm('+plus "in other term"')).to eq(['in other term', '+plus'])
-      expect(norm('with_blank \'\'')).to eq(%w[with_blank ''])
-      expect(norm('with_blank_doubles ""')).to eq(['with_blank_doubles'])
+      expect(norm(nil)).to eq({})
+      expect(norm('""')).to eq({})
+      expect(norm('-+""').pluck(:term)).to eq([])
+      expect(norm('simple words')).to eq('simple' => false, 'words' => false)
+      expect(norm(" with   \t spaces\n")).to eq('with' => false, 'spaces' => false)
+      expect(norm('with with duplicates with')).to eq('with' => false, 'duplicates' => false)
+      expect(norm('with "full term"')).to eq('full term' => false, 'with' => false)
+      expect(norm('"""odd double quotes around"""')).to eq('odd double quotes around' => false)
+      expect(norm('""even double quotes around""')).to eq('even double quotes around'=> false)
+      expect(norm('with\'apostrophe')).to eq("with'apostrophe" => false)
```

Comment:
can you pls add `'with-minus'` and `'with+plus'` as test cases too, they should both result in a single string result